signer/src/context/signer_state.rs

```diff
@@ -13,13 +14,98 @@ use crate::keys::PublicKey;
     pub fn current_signer_set(&self) -> &SignerSet {
         &self.current_signer_set
     }
+
+    /// Get the current sBTC limits.
+    pub fn get_current_limits(&self) -> SbtcLimits {
+        // We should never fail to acquire a lock from the RwLock so that it panics.
+        self.current_limits
+            .read()
+            .expect("BUG: Failed to acquire read lock")
+            .clone()
+    }
+
+    /// Update the current sBTC limits.
+    pub fn update_current_limits(&self, new_limits: SbtcLimits) {
+        // We should never fail to acquire a lock from the RwLock so that it panics.
+        let mut limits = self
+            .current_limits
+            .write()
+            .expect("BUG: Failed to acquire write lock");
+        *limits = new_limits;
+    }
+}
+
+/// Represents the current sBTC limits.
+#[derive(Debug, Clone, Default, PartialEq)]
+pub struct SbtcLimits {
+    /// Represents the total cap for all pegged-in BTC/sBTC.
+    total_cap: Option<Amount>,
+    /// Represents the maximum amount of BTC allowed to be pegged-in per transaction.
+    per_deposit_cap: Option<Amount>,
+    /// Represents the maximum amount of sBTC allowed to be pegged-out per transaction.
+    per_withdrawal_cap: Option<Amount>,
+    /// Represents the maximum amount of sBTC that can currently be minted.
+    max_mintable_cap: Option<Amount>,
+}
+
+impl std::fmt::Display for SbtcLimits {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "[total cap: {:?}, per-deposit cap: {:?}, per-withdrawal cap: {:?}, max-mintable cap: {:?}]",
+            self.total_cap, self.per_deposit_cap, self.per_withdrawal_cap, self.max_mintable_cap
+        )
+    }
+}
+
+impl SbtcLimits {
+    /// Create a new `SbtcLimits` object.
+    pub fn new(
+        total_cap: Option<Amount>,
+        per_deposit_cap: Option<Amount>,
+        per_withdrawal_cap: Option<Amount>,
+        max_mintable_cap: Option<Amount>,
+    ) -> Self {
+        Self {
+            total_cap,
+            per_deposit_cap,
+            per_withdrawal_cap,
+            max_mintable_cap,
+        }
+    }
+
+    /// Get the total cap for all pegged-in BTC/sBTC.
+    pub fn total_cap(&self) -> Option<Amount> {
+        self.total_cap
+    }
+
+    /// Get the maximum amount of BTC allowed to be pegged-in per transaction.
+    pub fn per_deposit_cap(&self) -> Option<Amount> {
+        self.per_deposit_cap
+    }
+
+    /// Get the maximum amount of sBTC allowed to be pegged-out per transaction.
+    pub fn per_withdrawal_cap(&self) -> Option<Amount> {
+        self.per_withdrawal_cap
+    }
+
+    /// Get the maximum amount of sBTC that can currently be minted.
+    pub fn max_mintable_cap(&self) -> Option<Amount> {
+        self.max_mintable_cap
+    }
+
+    /// Set the total cap for sBTC that can currently be minted.
+    pub fn set_max_mintable_cap(&mut self, max_mintable_cap: Option<Amount>) {
```

Comment:
When I had designed this I intended for this struct to be immutable, because when you pull it out you get a clone and any changes won't be reflected here. Just noting this, and if we keep this method here we should probably make it clear on the `get_current_limits` method that any changes made to the struct won't be reflected in global state.

Reply:
I removed the method and did what you proposed below